src/fair/metrics.py

```diff
@@ -299,3 +305,157 @@ def EF_X_agents(
                         envy_count += 1
                         break
     return envy_count
+
+
+# FUNCTIONS TO COMPUTE PAIRWISE MAXIMIN SHARE
+
+
+def create_sub_schedule(bundle_1: list[ScheduleItem], bundle_2: list[ScheduleItem]):
+    """Given two subsets of the set of items, create a new sub schedule considering the items in the union of both sets.
+    Capacities of the new schedule are determined by repetitions of the items in both bundles.
+    This function is necessary to compute PMMS metric. Creates a schedule with the items currently owned by two agents, in order to solve subproblem.
+
+    Args:
+        bundle_1 (list[ScheduleItem]): Items from class BaseItem
+        bundle_2 (list[ScheduleItem]): Items from class BaseItem
+
+    Returns:
+        new_schedule (list[ScheduleItem]): Items from class BaseItem, new reduced schedule
+        course_strings (list[str]): List of course strings of the new schedule
+        course (type[Course]): Course instance of the new schedule
+    """
+    sub_schedule = [*bundle_1, *bundle_2]
+    set_sub_schedule = sorted(list(set(sub_schedule)), key=lambda item: item.values[0])
+
+    course_strings = sorted([item.values[0] for item in set_sub_schedule])
+    course = Course(course_strings)
+    section = Section(sorted(list(set([item.values[3] for item in set_sub_schedule]))))
+    slot = sub_schedule[0].features[1]
+    weekday = sub_schedule[0].features[2]
+
+    features = [course, slot, weekday, section]
+
+    new_schedule = []
+    for i, item in enumerate(set_sub_schedule):
+        new_schedule.append(
+            ScheduleItem(
+                features, item.values, index=i, capacity=sub_schedule.count(item)
+            )
+        )
+    return new_schedule, course_strings, course
+
+
+def yankee_swap_sub_problem(
+    agent: type[BaseAgent],
+    new_schedule: list[ScheduleItem],
+    course_strings: list[str],
+    course: type[Course],
+):
+    """Given an agent and information of a reduced schedule (new_schedule, course_strings, course), compute their MMS for the reduced problem,
+    considering 2 identical agents competing for the items in the reduced schedule.
+    We do this by computing a leximin allocation through yankee swap.
+
+    Args:
+        agent (type[BaseAgent]): Agent from the class BaseAgent
+        new_schedule (list[ScheduleItem]): Items from class BaseItem, new reduced schedule
+        course_strings (list[str]): List of course strings of the new schedule
+        course (type[Course]): Course instance of the new schedule
+
+    Returns:
+        int: Agent's MMS for the subproblem
+    """
+    course_time_constr = CourseTimeConstraint.from_items(
+        new_schedule, new_schedule[0].features[1], new_schedule[0].features[2]
```

Comment:
It's better to pass to this method the features that you want to access instead of rummaging for it in the schedule. For example
```
course_time_constr = CourseTimeConstraint.from_items(new_schedule, course, section)
```